### ISSUE TITLE
Set polling length in waitForTransactionReceipt

### ISSUE THREAD

**Issue by carver** [ethereum/web3.py#1130]
### What was wrong?

When calling `w3.eth.waitForTransactionReceipt()` waiting for a transaction receipt, hitting every 100ms is a little too much for Infura. There is no way to change the polling interval.

### How can it be fixed?

Add a new `poll_interval` keyword arg to the method, and pass it along to `wait_for_transaction_receipt`. Write a test to confirm.

**Comment by abhilashreddyy** [comment]
I like to contribute to web3. Can I know if this is up for grabs.

**Comment by carver** [comment]
Yes, we would love to have you contribute. We encourage opening a PR very early and marking `[WIP]` (Work in progress) as an indicator that you know it's not ready to merge yet.

**Comment by abhilashreddyy** [comment]
```
def wait_for_transaction_receipt(web3, txn_hash, timeout=120, poll_latency=0.1):
    with Timeout(timeout) as _timeout:
        while True:
            txn_receipt = web3.eth.getTransactionReceipt(txn_hash)
            if txn_receipt is not None:
                break
            _timeout.sleep(poll_latency)
    return txn_receipt
```

I found that `poll_latency` is already added by someone. 
Did someone already solved this issue ?

**Comment by abhilashreddyy** [comment]
@carver  i have added `polling_interval` parameter to `waitForTransactionReceipt` function . Now i am trying to figure out where to write test and I am a little confused about that. Is there any directory in the code you can point me to, so that i can start figuring out where and how to write tests. It would be really helpful if you can list me out pre requisites that I need to know before starting contribution. Thankyou

**Comment by abhilashreddyy** [comment]
Hi @carver 
I have finished adding `poll_latency` parameter to `w3.eth.waitForTransactionReceipt()` function and wrote tests. This is my first opensource contribution .So, reply me if there any corrections.

**Comment by kclowes** [comment]
Fixed in #1453
